src/iosrv/vos.c

```diff
@@ -785,16 +786,17 @@ enum_unpack_recxs(daos_key_desc_t *kds, void *data,
 		return -DER_INVAL;
 
 	iod = &io->ui_iods[io->ui_iods_size - 1];
-	iod_akey = &iod->iod_name;
+	daos_iov_copy(&iod_akey, &iod->iod_name);
 	dkey = &io->ui_dkey;
-	if (dkey->iov_len == 0 || iod_akey->iov_len == 0) {
+	if (dkey->iov_len == 0 || iod_akey.iov_len == 0) {
 		rc = -DER_INVAL;
 		D_ERROR("invalid list buf %c\n", rc);
 		return rc;
```

Comment:
memory leak, actually I don't see why you change to allocating iod_akey in this patch.

Reply:
ah, ok